auth_jwt_test.go

```diff
@@ -325,3 +326,75 @@ func TestAuthJWTPayload(t *testing.T) {
 	}
 
 }
+
+func TestClaimsDuringAuthorization(t *testing.T) {
+	authMiddleware := &JWTMiddleware{
+		Realm:            "test zone",
+		SigningAlgorithm: "HS256",
+		Key:              key,
+		Timeout:          time.Hour,
+		MaxRefresh:       time.Hour * 24,
+		PayloadFunc: func(userId string) map[string]interface{} {
+			// Set custom claim, to be checked in Authorizator method
+			return map[string]interface{}{"testkey": "testval", "exp": 0}
+		},
+		Authenticator: func(userId string, password string) bool {
+			log.Printf("Authentication Attempt: %s\n", userId)
+			return true
+		},
+		Authorizator: func(userId string, request *rest.Request) bool {
+			// Check that payload is set, output helpful debugging message if not
+			if request.Env["JWT_PAYLOAD"] == nil {
+				log.Println("JWT_PAYLOAD was unset during Authorization!")
+				return false
+			}
+			// Cast / Extract claims
+			jwt_claims := request.Env["JWT_PAYLOAD"].(map[string]interface{})
+
+			// Check the actual claim, set in PayloadFunc
+			return (jwt_claims["testkey"] == "testval")
```

Comment:
I think we should also test that `request.Env["REMOTE_USER"]` is available. Maybe AND the return statement with `request.Env["REMOTE_USER"] == "admin"`